lib/github.server-action.ts

```diff
@@ -0,0 +1,55 @@
+import { IGitHubEvent } from "@/lib/gh_events";
+import { Octokit } from "octokit";
+
+const GITHUB_PAT: string = process.env.GITHUB_PAT || "";
+const GITHUB_ORG: string = process.env.NEXT_PUBLIC_GITHUB_ORG || "";
+
+const octokit = new Octokit({
+  auth: GITHUB_PAT,
+});
+
+export const eventTypes = [
+  "MemberEvent",
+  "IssuesEvent",
+  "PullRequestEvent",
+  "PullRequestReviewEvent",
+  "PushEvent",
+  "ForkEvent",
+  "ReleaseEvent",
+  "IssueCommentEvent",
+];
+
+export async function fetchFilteredGitHubEvents() {
+  try {
+    const response = await octokit.request("GET /orgs/{org}/events", {
```

Comment:
Do a `.paginate` instead as that'll fetch all the available pages. You can filter the items inside the callback itself to reduce maemory usage